snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -157,6 +158,15 @@ def _set_image_info_env_var(self):
         for field in ('fingerprint', 'architecture', 'created_at'):
             if field in image_info[0]:
                 edited_image_info[field] = image_info[0][field]
+
+        # Pick up existing image info if set
+        image_info_str = os.environ.get('SNAPCRAFT_IMAGE_INFO')
+        if image_info_str:
+            try:
+                edited_image_info.update(json.loads(image_info_str))
+            except json.decoder.JSONDecodeError as e:
+                raise InvalidContainerImageInfoError(image_info_str) from e
```

Comment:
This error says nothing of the expectation that this is JSON. I say we fix that here.

Reply:
Good call. Updated the error with a nicer message.